### PR TITLE
DCS-1280 add controller, service and view for prison transfer

### PR DESCRIPTION
**Pull request by GurnankCheema**
Integration test to be added in another ticket

view looks like this

transfer added to prison roll 26:11:21

### CODE REVIEW

File: server/routes/confirmTransferAddedToRollController.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    private readonly transfersService: TransfersService
+  ) {}
+
+  public view(): RequestHandler {
+    return async (req, res) => {
+      const { prisonNumber } = req.params
+      const { activeCaseLoadId } = res.locals.user
+      const { firstName, lastName } = req.flash('prisoner')?.[0] as Record<string, string>
+
+      if (!firstName && !lastName) {
```
Comment (GurnankCheema) [inline server/routes/confirmTransferAddedToRollController.ts:17]: there are people who only have a one name, i.e. just a firstName!

File: server/routes/confirmTransferAddedToRollController.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        return res.redirect('/confirm-arrival/choose-prisoner')
+      }
```
Comment (GurnankCheema) [inline server/routes/confirmTransferAddedToRollController.ts:19]: there are people who only have just one name i.e.a firstName!

File: server/routes/checkTransferController.ts [hunk 3]
```diff
@@ -16,8 +17,24 @@ export default class CheckTransferController {
...
+        `AgencyId: ${activeCaseLoadId}, From: ${data.fromLocation}, Type: 'PRISON',`,
+        req.hostname
+      )
+
+      req.flash('prisoner', {
+        firstName: data.firstName,
+        lastName: data.lastName,
+      })
+
+      await this.transfersService.confirmTransfer(username, prisonNumber)
```
Comment (andrewrlee) [inline server/routes/checkTransferController.ts:36]: maybe we should swap round the raiseAnalyticsEvent so it only happens in the confirmTransfer was successful?

Comment (andrewrlee) [reply server/routes/checkTransferController.ts:36]: we should probably do the same for confirmArrival also if its been done that way

Comment (GurnankCheema) [reply server/routes/checkTransferController.ts:36]: ok

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**